examples/main/main.cpp

```diff
@@ -319,6 +316,10 @@ int main(int argc, char ** argv) {
                 embd.insert(embd.begin(), last_n_tokens.begin() + n_ctx - n_left/2 - embd.size(), last_n_tokens.end() - embd.size());
 
                 // stop saving session if we run out of context
+                if (!path_session.empty() && session_save_all) {
+                    llama_save_session_file(ctx, path_session.c_str(),
+                        session_tokens.data(), session_tokens.size());
+                }
```

Comment:
Is there a use case to save the session when the context is full?
When we reload the session, it will be re-evaluated anyway due to the "context swap" logic.

Maybe remove this if it is not needed and merge it after addressing @DannyDaemonic comments